manager/controllers/clusters/cves.go

```diff
@@ -0,0 +1,141 @@
+// @Param cvss_score      query []number false "CVSS score of CVE: (from float),(to float)" collectionFormat(multi) collectionFormat(csv) minItems(2) maxItems(2)
+// @Param affected_images query []bool 	 false "minimal count of affected images: (1 or more),(none)" collectionFormat(multi) collectionFormat(csv) minItems(2) maxItems(2)
+// @router /clusters/{cluster_id}/cves [get]
+// @success 200 {object} base.Response{data=GetClusterCvesSelect}
+// @failure 400 {object} base.Error
+// @failure 404 {object} base.Error "cluster does not exist"
+// @failure 500 {object} base.Error
+func (c *Controller) GetClusterCves(ctx *gin.Context) {
+	accountID := ctx.GetInt64("account_id")
+	clusterID, err := base.GetParamUUID(ctx, "cluster_id")
+	if err != nil {
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, base.BuildErrorResponse(http.StatusBadRequest, err.Error()))
+		return
+	}
+
+	exists, err := c.ClusterExists(accountID, clusterID)
+	if err != nil {
+		ctx.AbortWithStatusJSON(
+			http.StatusInternalServerError,
+			base.BuildErrorResponse(http.StatusInternalServerError, "Internal server error"),
+		)
+		return
+	} else if !exists {
+		ctx.AbortWithStatusJSON(
+			http.StatusBadRequest,
+			base.BuildErrorResponse(http.StatusNotFound, "cluster does not exist"),
+		)
+		return
+	}
+
+	filters := base.GetRequestedFilters(ctx)
+	query := c.BuildClusterCvesQuery(accountID, clusterID)
+
+	err = base.ApplyFilters(query, getClusterCvesAllowedFilters, filters, getClusterCvesFilterArgs)
+	if err != nil {
+		ctx.AbortWithStatusJSON(
+			http.StatusBadRequest,
+			base.BuildErrorResponse(http.StatusBadRequest, err.Error()),
+		)
+		return
+	}
+	dataRes := []GetClusterCvesSelect{}
+	res := query.Find(&dataRes)
+	if res.Error != nil {
+		ctx.AbortWithStatusJSON(
+			http.StatusInternalServerError,
+			base.BuildErrorResponse(http.StatusInternalServerError, "Internal server error"),
+		)
+		return
+	}
+
+	ctx.JSON(http.StatusOK,
+		base.BuildResponse(dataRes, base.BuildMeta(filters, getClusterCvesAllowedFilters)),
+	)
+}
+
+func (c *Controller) ClusterExists(accountID int64, clusterID uuid.UUID) (bool, error) {
+	res := c.Conn.Table("cluster").Where("account_id = ? AND uuid = ?", accountID, clusterID).Limit(1).Find(&struct{}{})
+	if res.Error != nil {
+		return false, res.Error
+	}
+	return res.RowsAffected > 0, nil
+}
+
+func (c *Controller) BuildClusterCvesQuery(accountID int64, clusterID uuid.UUID) *gorm.DB {
+	subquery := c.Conn.Table("cluster_image").
+		Select("cluster_image.image_id").
+		Joins("JOIN cluster ON cluster_image.cluster_id = cluster.id").
+		Where("cluster.account_id = ? AND cluster.uuid = ?", accountID, clusterID.String())
+
+	return c.Conn.Table("cve").
+		Select(`cve.cvss2_score, cve.cvss3_score, cve.description, cve.severity,
+			cve.public_date, cve.name,
+			COUNT(DISTINCT cluster_image.image_id) as images_exposed`).
+		Joins("LEFT JOIN image_cve ON cve.id = image_cve.cve_id").
```

Comment:
shouldn't be INNER join to return only affected CVEs?

Reply:
We want to support `affected_images` filter, thus left join if customer wants to see cves with 0 affected images.

Reply:
Reworked it so CVEs with 0 images exposed are not returned and all and removed the filter from openapi.